components/finance/ExpensePanelRows/index.tsx

```diff
@@ -0,0 +1,114 @@
+import { createStyles, FormControl, makeStyles, MenuItem, Select, TextField } from '@material-ui/core'
+import React from 'react'
+
+const useStyles = makeStyles(() =>
+  createStyles({
+    formControl: {
+      backgroundColor: 'white',
+      margin: '0.5rem',
+      width: '4rem',
+      height: '2.5rem',
+    },
+    textField: {
+      backgroundColor: 'white',
+      borderRadius: 'rounded',
+      marginTop: '0.5rem',
+      marginBottom: '0.5rem',
+      marginRight: '0.5rem',
+    },
+    amountField: {
+      width: '8rem',
+      backgroundColor: 'white',
+      borderRadius: 'rounded',
+      marginTop: '0.5rem',
+      marginBottom: '0.5rem',
+      marginRight: '0.5rem',
+    },
+  })
+)
+
+export default function ExpensePanelRows({
+  totalValues,
+  handleChange,
+  handleRemoveSpecificRow,
+  currencyList,
+  defaultCurrency,
+}: any) {
+  const classes = useStyles()
+  return (
+    <div>
+      {totalValues.rows.map((item: any, idx: any) => (
+        <div id={idx} key={idx.toString()}>
+          <FormControl variant="outlined">
+            <Select
+              className={classes.formControl}
+              placeholder="Currency"
+              name="currency"
+              value={totalValues.rows[idx].currency === undefined ? defaultCurrency : totalValues.rows[idx].currency}
+              onChange={handleChange(idx)}
+            >
+              {currencyList.map((currency: string, currencyIdx: any) => (
+                <MenuItem id={currencyIdx} value={currency} key={currencyIdx.toString()}>
+                  {currency}
+                </MenuItem>
+              ))}
+            </Select>
+          </FormControl>
+          <TextField
+            value={totalValues.rows[idx].amount === undefined ? 0 : totalValues.rows[idx].amount}
+            InputProps={{
+              inputProps: {
+                max: 100,
+                min: 1,
+                step: 0.01,
+              },
+            }}
+            onChange={handleChange(idx)}
+            className={classes.amountField}
+            placeholder="Amount"
+            type="number"
+            name="amount"
+            margin="dense"
+            variant="outlined"
+          />
+          <TextField
+            value={totalValues.rows[idx].comment}
+            onChange={handleChange(idx)}
+            className={classes.textField}
+            name="comment"
+            placeholder="Comment"
+            margin="dense"
+            variant="outlined"
+          />
+          <TextField
```

Comment:
use the TagControl implemented by Aryaman

Reply:
can't find it in repo yet